pkg/river/json_walker.go

```diff
@@ -0,0 +1,409 @@
+package river
```

Comment:
I don't think this logic belongs in the top-level river package. Maybe `pkg/river/encoding/riverjson`?